stream_alert/shared/alert.py

```diff
@@ -152,27 +175,34 @@ def create_from_dynamo_record(cls, record):
 
         Returns:
             Alert: An alert with all properties populated from the Dynamo item
+
+        Raises:
+            AlertCreationError: If the record isn't formatted correctly.
         """
-        return cls(
-            record['RuleName'],
-            json.loads(record['Record']),
-            record['Outputs'],
-            alert_id=record['AlertID'],
-            attempts=record.get('Attempts'),
-            cluster=record.get('Cluster'),
-            context=record.get('Context'),
-            created=record.get('Created'),
-            dispatched=record.get('Dispatched'),
-            log_source=record.get('LogSource'),
-            log_type=record.get('LogType'),
-            merge_by_keys=record.get('MergeByKeys'),
-            merge_window_mins=record.get('MergeWindowMins'),
-            retry_outputs=record.get('RetryOutputs'),
-            rule_description=record.get('RuleDescription'),
-            source_entity=record.get('SourceEntity'),
-            source_service=record.get('SourceService'),
-            staged=record.get('Staged')
-        )
+        try:
+            return cls(
+                record['RuleName'],
+                json.loads(record['Record']),
+                set(record['Outputs']),  # In JSON, outputs may be a list - convert back to set
+                alert_id=record['AlertID'],
+                attempts=record.get('Attempts'),
+                cluster=record.get('Cluster'),
+                context=record.get('Context'),
+                created=datetime.strptime(record['Created'], cls.DATETIME_FORMAT),
+                dispatched=datetime.strptime(
+                    record['Dispatched'], cls.DATETIME_FORMAT) if 'Dispatched' in record else None,
+                log_source=record.get('LogSource'),
+                log_type=record.get('LogType'),
+                merge_by_keys=record.get('MergeByKeys'),
+                merge_window=timedelta(minutes=int(record.get('MergeWindowMins', 0))),
+                outputs_sent=set(record.get('OutputsSent') or []),
+                rule_description=record.get('RuleDescription'),
+                source_entity=record.get('SourceEntity'),
+                source_service=record.get('SourceService'),
+                staged=record.get('Staged')
+            )
+        except (KeyError, TypeError, ValueError) as error:
+            raise AlertCreationError(error)
```

Comment:
good thinking!

Reply:
Thanks. I wasn't sure if this was overkill or not, but I kept catching the same tuple of errors and figured it might be nice if it had a name